### PR TITLE
Add metamonitoring docs

### PR DESCRIPTION
**Pull request by dimitarvdimitrov**
Signed-off-by: Dimitar Dimitrov <dimitar.dimitrov@grafana.com>

#### What this PR does

Moves metamonitoring docs from internal docs for Grafana Cloud integration to Mimir docs.

Also enhances the information for the requirements on logs and metrics labels.

#### Which issue(s) this PR fixes or relates to

Fixes https://github.com/grafana/mimir/issues/2332

### CODE REVIEW

File: docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+menuTitle: "Collecting metrics and logs"
+description: "Learn how to collect metrics and logs from Grafana Mimir itself"
+weight: 60
+---
+
+# Collecting metrics and logs from Grafana Mimir
+
+This article helps you to collect logs and metrics from a Mimir or GEM cluster. To set up dashboards and alerts,
+see [Installing Grafana Mimir dashboards and alerts]({{< relref "installing-dashboards-and-alerts.md" >}})
+or [Grafana Cloud: Self-hosted Grafana Mimir integration](https://grafana.com/docs/grafana-cloud/integrations/integrations/integration-mimir/)
```
Comment (osg-grafana) [inline docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md:12]:
```suggestion
You can collect logs and metrics from a Mimir or GEM cluster. To set up dashboards and alerts,
see [Installing Grafana Mimir dashboards and alerts]({{< relref "installing-dashboards-and-alerts.md" >}})
or [Grafana Cloud: Self-hosted Grafana Mimir integration](https://grafana.com/docs/grafana-cloud/integrations/integrations/integration-mimir/)
```

File: docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+
+* For a stable release:
+    * \>= 3.x.x:
+      See [Collect metrics and logs via Helm chart built-in metamonitoring](#collect-metrics-and-logs-via-the-helm-chart)
+    * \< 3.x.x: See [Collect metrics and logs via Grafana Agent](#collect-metrics-and-logs-via-grafana-agent)
+* For non-Helm installations or installations of the deprecated enterprise-metrics Helm chart, see [Collect metrics and logs without the helm chart](#collect-metrics-and-logs-without-the-helm-chart).
+
+### Collect metrics and logs via the Helm chart
+
+Starting from version `3.0.0`, the Helm chart sends metrics to a Prometheus-compatible server and logs to a Loki
```
[the remark this answers is not in the document]
Comment (dimitarvdimitrov) [reply docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md:33]:
```suggestion
Starting from version `3.0.0`, the Helm chart sends metrics to a Prometheus-compatible server and sends logs to a Loki
```

File: docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+cluster. The chart can also scrape additional metrics from kube-state-metrics, kubelet, and cAdvisor. node_exporter
```
Comment (osg-grafana) [inline docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md:34]:
```suggestion
cluster. The chart can also scrape additional metrics from kube-state-metrics, kubelet, and cAdvisor. Any node_exporter
```
Adding a word in front of node_exporter so the sentence does not start with a lowercase letter.

File: docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+metrics are not collected by the chart.
+See [Additional resources metrics]({{< relref "requirements.md#additional-resources-metrics" >}}) for more information
+about node_exporter.
```
Comment (osg-grafana) [inline docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md:37]:
```suggestion
For more information
about node_exporter, see [Additional resources metrics]({{< relref "requirements.md#additional-resources-metrics" >}}).
```

File: docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md (hunk 3 above)
Comment (dimitarvdimitrov) [reply docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md:34]: Will change to "No node_exporter metrics are collected by the chart."

File: docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+The Helm chart uses the Grafana Agent operator. Due to how Helm works, before it can use the operator, you need to
+manually install
+the [Custom Resource Definitions (CRDs)](https://github.com/grafana/agent/tree/main/production/operator/crds) for the
+Agent operator.
+
+#### Credentials
+
+If Prometheus and Loki are running without authentication, then you scan skip this section.
+Metamonitoring supports multiple ways of authentication for metrics and logs. If you are using a secret (e.g. an API
+key) to authenticate with Prometheus or Loki, then you need to create a Kubernetes secret with that secret.
```
Comment (osg-grafana) [inline docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md:48]:
```suggestion
Metamonitoring supports multiple ways of authentication for metrics and logs. If you are using a secret such as an API
key to authenticate with Prometheus or Loki, then you need to create a Kubernetes secret with that secret.
```

File: docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md [hunk 6]
```diff
@@ -0,0 +1,335 @@
...
+kind: Secret
+metadata:
+  name: metamonitoring-credentials
+data:
+  prometheus-api-key: FAKEACCESSKEY
+  loki-api-key: FAKESECRETKEY
+```
+
+For information about how to create a Kubernetes secret, see
+the [Creating a Secret](https://kubernetes.io/docs/concepts/configuration/secret/#creating-a-secret).
```
Comment (osg-grafana) [inline docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md:63]:
```suggestion
[Creating a Secret](https://kubernetes.io/docs/concepts/configuration/secret/#creating-a-secret).
```

File: docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md [hunk 7]
```diff
@@ -0,0 +1,335 @@
...
+          namespace: kube-system
+          labelSelectors:
+            app.kubernetes.io/name: kube-state-metrics
+```
+
+### Collect metrics and logs via Grafana Agent
+
+Older versions of the Helm chart need to be manually instrumented. This means that you need to set up a Grafana Agent
+that collects logs and metrics from Mimir or GEM. Refer
+to [Set up Grafana Agent](https://grafana.com/docs/agent/latest/set-up/) for instructions on setting up the Agent.
```
Comment (osg-grafana) [inline docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md:107]:
```suggestion
that collects logs and metrics from Mimir or GEM. For instructions on setting up the Agent,
see [Set up Grafana Agent](https://grafana.com/docs/agent/latest/set-up/).
```

File: docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md [hunk 8]
```diff
@@ -0,0 +1,335 @@
...
+
+Below is an Agent configuration file for collecting logs and metrics. Replace `url`, `password`, and `username` in
+the `logs` and `metrics` blocks with the details of your Prometheus and Loki clusters.
```
Comment (osg-grafana) [inline docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md:110]:
```suggestion
In the following example Agent configuration file for collecting logs and metrics, replace `url`, `password`, and `username` in
the `logs` and `metrics` blocks with the details of your Prometheus and Loki clusters.
```
Agent == Grafana Agent or something else?

File: docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md [hunk 9]
```diff
@@ -0,0 +1,335 @@
...
+  global:
+    scrape_interval: 15s
+  wal_directory: /tmp/grafana-agent-wal
+server:
+  http_listen_port: 12345
+```
+
+## Collect metrics and logs without the Helm chart
+
+You can still use the dashboards and rules in the monitoring-mixin even if Mimir or GEM is not deployed via the Helm
```
Comment (osg-grafana) [inline docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md:321]:
```suggestion
You can still use the dashboards and rules in the monitoring-mixin, even if Mimir or GEM is not deployed via the Helm
```

File: docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md [hunk 10]
```diff
@@ -0,0 +1,335 @@
...
+chart or if you are using the deprecated enterprise-metrics Helm chart for GEM.
```
Comment (osg-grafana) [inline docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md:322]:
```suggestion
chart, or if you are using the deprecated enterprise-metrics Helm chart for GEM.
```

File: docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md [hunk 11]
```diff
@@ -0,0 +1,335 @@
...
+As a starting point, use the Agent configuration
+from [Collect metrics and logs via Grafana Agent](#collect-metrics-and-logs-via-grafana-agent) as a starting point.
```
Comment (osg-grafana) [inline docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md:324]:
```suggestion
from [Collect metrics and logs via Grafana Agent](#collect-metrics-and-logs-via-grafana-agent).
```

File: docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md [hunk 12]
```diff
@@ -0,0 +1,335 @@
...
+There may be some
+modifications you need to make to it. See [dashboards and alerts requirements]({{< relref "requirements.md" >}}) for
+more information.
```
Comment (osg-grafana) [inline docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md:327]:
```suggestion
You might need to modify it. For
more information, see [dashboards and alerts requirements]({{< relref "requirements.md" >}}).
```

File: docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md [hunk 13]
```diff
@@ -0,0 +1,335 @@
...
+
+### Service discovery
+
+The Agent configuration relies on Kubernetes service discovery and pod labels to constrain the collected metrics and
```
Comment (osg-grafana) [inline docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md:331]: Is this "pod" or "pods’"

File: docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md [hunk 14]
```diff
@@ -0,0 +1,335 @@
...
+logs to ones strictly related to the Helm chart. If you are deploying Grafana Mimir on something other than Kubernetes,
```
Comment (osg-grafana) [inline docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md:332]:
```suggestion
logs to ones that are strictly related to the Helm chart. If you are deploying Grafana Mimir on something other than Kubernetes,
```

File: docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md [hunk 15]
```diff
@@ -0,0 +1,335 @@
...
+then you
+need to replace the `kubernetes_sd_configs` block with a block from
```
Comment (osg-grafana) [inline docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md:334]:
```suggestion
then replace the `kubernetes_sd_configs` block with a block from
```

File: docs/sources/operators-guide/monitoring-grafana-mimir/requirements.md [hunk 16]
```diff
@@ -55,9 +55,25 @@ When running Grafana Mimir in microservices mode and using the pre-compiled dash
...
-- kubelet
+- [cAdvisor](https://github.com/google/cadvisor)
+- [kubelet](https://kubernetes.io/docs/concepts/cluster-administration/system-metrics/)
 - [node_exporter](https://github.com/prometheus/node_exporter)
 - [kube-state-metrics](https://github.com/kubernetes/kube-state-metrics) exporter
 
 For more information about the kubelet metrics and cAdvisor metrics exported by the kubelet, refer to [Metrics For Kubernetes System Components](https://kubernetes.io/docs/concepts/cluster-administration/system-metrics/).
+
+Metrics from kubelet, kube-state-metrics, and cAdvisor should all have a `cluster` label with the same value as in the
+Mimir metrics.
```
Comment (osg-grafana) [inline docs/sources/operators-guide/monitoring-grafana-mimir/requirements.md:66]:
```suggestion
Metrics from kubelet, kube-state-metrics, and cAdvisor must all have a `cluster` label with the same value as in the
Mimir metrics.
```

File: docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md (hunk 13 above)
Comment (dimitarvdimitrov) [reply docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md:331]: "pod labels" = "labels of pods"

I think it is "pod", not "pods"

File: docs/sources/operators-guide/monitoring-grafana-mimir/requirements.md [hunk 17]
```diff
@@ -55,9 +55,25 @@ When running Grafana Mimir in microservices mode and using the pre-compiled dash
...
+
+Metrics from node_exporter should all have an `instance` label on them that has the same value as the `instance` label on Mimir metrics.
```
[the remark this answers is not in the document]
Comment (dimitarvdimitrov) [reply docs/sources/operators-guide/monitoring-grafana-mimir/requirements.md:68]: must be must

```suggestion
Metrics from node_exporter must all have an `instance` label on them that has the same value as the `instance` label on Mimir metrics.
```

File: docs/sources/operators-guide/monitoring-grafana-mimir/requirements.md [hunk 18]
```diff
@@ -55,9 +55,25 @@ When running Grafana Mimir in microservices mode and using the pre-compiled dash
...
+
+## Log labels
+
+The "Slow Queries" dashboard uses a Loki datasource with the logs from Grafana Mimir to visualize slow queries. The query-frontend component logs slow queries as configured by the `-query-frontend.log-queries-longer-than` parameter.
+These logs need to have certain labels in order for the dashboard to work.
```
Comment (osg-grafana) [inline docs/sources/operators-guide/monitoring-grafana-mimir/requirements.md:73]:
```suggestion
The **Slow queries** dashboard uses a Loki data source with the logs from Grafana Mimir to visualize slow queries. The query-frontend component logs slow queries based on how you configured the `-query-frontend.log-queries-longer-than` parameter.
These logs need to have specific labels in order for the dashboard to work.
```
Change dashboard to sentence case upstream if need be.

File: docs/sources/operators-guide/monitoring-grafana-mimir/requirements.md [hunk 19]
```diff
@@ -55,9 +55,25 @@ When running Grafana Mimir in microservices mode and using the pre-compiled dash
...
+
+| Label Name  | Configurable | Description                                                                                                                                                          |
```
Comment (osg-grafana) [inline docs/sources/operators-guide/monitoring-grafana-mimir/requirements.md:75]:
```suggestion
| Label name  | Configurable | Description                                                                                                                                                          |
```

File: docs/sources/operators-guide/monitoring-grafana-mimir/requirements.md (hunk 18 above)
Comment (dimitarvdimitrov) [reply docs/sources/operators-guide/monitoring-grafana-mimir/requirements.md:73]: it's already sentence case in the dashboard 👍

File: docs/sources/operators-guide/monitoring-grafana-mimir/requirements.md [hunk 20]
```diff
@@ -55,9 +55,25 @@ When running Grafana Mimir in microservices mode and using the pre-compiled dash
...
+| :---------- | :----------- | :------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `cluster`   | Yes          | The Kubernetes cluster or datacenter where the Mimir cluster is running. The cluster label can be configured with the `per_cluster_label` field in the mixin config. |
```
Comment (osg-grafana) [inline docs/sources/operators-guide/monitoring-grafana-mimir/requirements.md:77]:
```suggestion
| `cluster`   | Yes          | The Kubernetes cluster or datacenter where the Mimir cluster is running. You can configure the cluster label via the `per_cluster_label` field in the mixin configuration. |
```

File: docs/sources/operators-guide/monitoring-grafana-mimir/requirements.md [hunk 21]
```diff
@@ -55,9 +55,25 @@ When running Grafana Mimir in microservices mode and using the pre-compiled dash
...
+| `namespace` | No           | The Kubernetes namespace where the Mimir cluster is running.                                                                                                         |
+| `name`      | No           | Name of the component. e.g. `query-frontend`                                                                                                                         |
```
Comment (osg-grafana) [inline docs/sources/operators-guide/monitoring-grafana-mimir/requirements.md:79]:
```suggestion
| `name`      | No           | Name of the component. For example, `query-frontend`.                                                                                                                         |
```

**Review by osg-grafana — APPROVED** [review]
Unblocking with several nits.

File: docs/sources/operators-guide/monitoring-grafana-mimir/requirements.md (hunk 20 above)
Comment (dimitarvdimitrov) [reply docs/sources/operators-guide/monitoring-grafana-mimir/requirements.md:77]: will also change the previous occurrence in the file

File: docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md (hunk 13 above)
Comment (osg-grafana) [reply docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md:331]: I am likely over thinking this. Keep as is.

File: docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md (hunk 3 above)
Comment (osg-grafana) [reply docs/sources/operators-guide/monitoring-grafana-mimir/collecting-metrics-and-logs.md:34]: Alternatively, The Helm chart does not collect node_exporter metrics.

**Comment by osg-grafana** [comment]
Excellent work, @dimitarvdimitrov! :)

**Pull request merged**